src/java_bytecode/java_bytecode_typecheck_expr.cpp

```diff
@@ -208,26 +228,38 @@ Function: java_bytecode_typecheckt::typecheck_expr_symbol
 
 void java_bytecode_typecheckt::typecheck_expr_member(member_exprt &expr)
 {
-  // The member might be in a parent class, which we resolve here.
+  // The member might be in a parent class or an opaque class, which we resolve here.
   const irep_idt component_name=expr.get_component_name();
 
   while(1)
   {
-    if(ns.follow(expr.struct_op().type()).id()!=ID_struct)
+
+    typet &base_type = const_cast<typet&>(ns.follow(expr.struct_op().type()));
```

Comment:
in the case of an opaque stub, there will be a component added to the `struct_type` of this `base_type`, which prevents having `const` here